### ISSUE TITLE
Think about ways to reduce amount of method definition overhead per move

### ISSUE THREAD

**Issue by jkomoros** [jkomoros/boardgame#305]
Currently it's a lot of overhead because moves are called for a LOT

A few options:

1. Some kind of codegen approach
1. A DefaultMove that you can embed so you only have to implement the things that are non-default. (Does that get weird with PropertyReader? I don't think so...)
1. Something crazy like having Move be a concrete struct that takes a delegate or something

2 is almost certainly the right approach

**Comment by jkomoros** [comment]
```
type DefaultMove struct {
  Name string
  Description string
}

// no-op versions of all move interface required follow

type MyMove struct {
  boardgame.DefaultMove
  MyIntProp int
  MyStackProp
}
```

This is way harder to do without some kind of automated Copy thing like in #297 (everyone has to override Copy, and it's still error-prone)

**Comment by jkomoros** [comment]
If we wanted, we could require that all Moves embed BaseMove as first slot... or we could have it be purely an optional convenience...

**Comment by jkomoros** [comment]
Copy could take a reference to the manager, get the default move by that name, then copy over all properties that property reader can see...

**Comment by jkomoros** [comment]
If we wanted to get rid of the Copy() method, which requires eveyrone to implement their own, we could have the thing you add to a game be a MoveFactory() or something, MoveForState() that returns a fresh version each time.

**Comment by jkomoros** [comment]
We can't implement Copy or ReadSetter, maybe need a code-gen solution for that. :-/

**Comment by jkomoros** [comment]
DefaultMove's DefaultsForState should set all PlayerIndexes to CurrentPlayer

**Comment by jkomoros** [comment]
If we resolve #297 for Moves as well then it's only one annoying method, the PropertyReadSetter interface. And if we solve #301 then even that one goes away.

**Comment by jkomoros** [comment]
Now that we fixed #308, the only annoying one left is ReadSetter, which will be handled by #301, so we can just close this issue
